### ISSUE TITLE
Incorrect Bug Reports

### ISSUE THREAD

**Issue by audxo14** [ConsenSys/mythril-classic#389]
## Description
Mythril reports the wrong locations of the alarms regardless of its type.
(similar to the previous bug report - https://github.com/ConsenSys/mythril/issues/376).
Before I actually run Mythril, I deleted all the comments first and tested the file. 

## How to Reproduce
I used the docker environment as I failed to install Mythril on my computer.
The command:
sudo docker run $(pwd):/tmp mythril/myth -x /tmp/file.sol

The file.sol is the source code in:
https://etherscan.io/address/0xa646e29877d52b9e2de457eca09c724ff16d0a2b#code

The report is as follows:
```
==== Integer Overflow ====
Type: Warning
Contract: MultiSigWallet
Function name: _function_0xa0e67e2b
PC address: 1358
A possible integer overflow exists in the function `_function_0xa0e67e2b`.
The addition or multiplication may result in a value higher than the maximum representable integer.
--------------------
In file: /tmp/44.sol:259

function getOwners()
        public
        constant
        returns (address[])
    {
        return owners;
    }

--------------------

==== Integer Overflow ====
Type: Warning
Contract: MultiSigWallet
Function name: _function_0xa8abe69a
PC address: 1522
A possible integer overflow exists in the function `_function_0xa8abe69a`.
The addition or multiplication may result in a value higher than the maximum representable integer.
--------------------
In file: /tmp/44.sol:285

function getTransactionIds(uint from, uint to, bool pending, bool executed)
        public
        constant
        returns (uint[] _transactionIds)
    {
        uint[] memory transactionIdsTemp = new uint[](transactionCount);
        uint count = 0;
        uint i;
        for (i=0; i<transactionCount; i++)
            if (   pending && !transactions[i].executed
                || executed && transactions[i].executed)
            {
                transactionIdsTemp[count] = i;
                count += 1;
            }
        _transactionIds = new uint[](to - from);
        for (i=from; i<to; i++)
            _transactionIds[i - from] = transactionIdsTemp[i];
    }

--------------------

==== Integer Overflow ====
Type: Warning
Contract: MultiSigWallet
Function name: _function_0xb5dc40c3
PC address: 1652
A possible integer overflow exists in the function `_function_0xb5dc40c3`.
The addition or multiplication may result in a value higher than the maximum representable integer.
--------------------
In file: /tmp/44.sol:267

function getConfirmations(uint transactionId)
        public
        constant
        returns (address[] _confirmations)
    {
        address[] memory confirmationsTemp = new address[](owners.length);
        uint count = 0;
        uint i;
        for (i=0; i<owners.length; i++)
            if (confirmations[transactionId][owners[i]]) {
                confirmationsTemp[count] = owners[i];
                count += 1;
            }
        _confirmations = new address[](count);
        for (i=0; i<count; i++)
            _confirmations[i] = confirmationsTemp[i];
    }

--------------------

==== Exception state ====
Type: Informational
Contract: MultiSigWallet
Function name: _function_0x025e7c27
PC address: 2249
A reachable exception (opcode 0xfe) has been detected. This can be caused by type errors, division by zero, out-of-bounds array access, or assert violations. This is acceptable in most situations. Note however that `assert()` should only be used to check invariants. Use `require()` for regular input checking. 
--------------------
In file: /tmp/44.sol:20

address[] public owners

--------------------

==== Integer Underflow ====
Type: Warning
Contract: MultiSigWallet
Function name: _function_0x173825d9
PC address: 2548
A possible integer underflow exists in the function `_function_0x173825d9`.
The subtraction may result in a value < 0.
--------------------
In file: /tmp/44.sol:126

owners.length - 1

--------------------

==== Exception state ====
Type: Informational
Contract: MultiSigWallet
Function name: _function_0x173825d9
PC address: 2592
A reachable exception (opcode 0xfe) has been detected. This can be caused by type errors, division by zero, out-of-bounds array access, or assert violations. This is acceptable in most situations. Note however that `assert()` should only be used to check invariants. Use `require()` for regular input checking. 
--------------------
In file: /tmp/44.sol:127

owners[i]

--------------------

==== Integer Overflow ====
Type: Warning
Contract: MultiSigWallet
Function name: _function_0x7065cb48
PC address: 3801
A possible integer overflow exists in the function `_function_0x7065cb48`.
The addition or multiplication may result in a value higher than the maximum representable integer.
--------------------
In file: /tmp/44.sol:113

owners.length + 1

--------------------

==== Integer Overflow ====
Type: Warning
Contract: MultiSigWallet
Function name: _function_0xa0e67e2b
PC address: 4835
A possible integer overflow exists in the function `_function_0xa0e67e2b`.
The addition or multiplication may result in a value higher than the maximum representable integer.
--------------------
In file: /tmp/44.sol:264

return owners

--------------------

==== Integer Overflow ====
Type: Warning
Contract: MultiSigWallet
Function name: _function_0xa0e67e2b
PC address: 4837
A possible integer overflow exists in the function `_function_0xa0e67e2b`.
The addition or multiplication may result in a value higher than the maximum representable integer.
--------------------
In file: /tmp/44.sol:264

return owners

--------------------

==== Integer Underflow ====
Type: Warning
Contract: MultiSigWallet
Function name: _function_0xa8abe69a
PC address: 5163
A possible integer underflow exists in the function `_function_0xa8abe69a`.
The subtraction may result in a value < 0.
--------------------
In file: /tmp/44.sol:300

to - from

--------------------

==== Exception state ====
Type: Informational
Contract: MultiSigWallet
Function name: _function_0xa8abe69a
PC address: 5237
A reachable exception (opcode 0xfe) has been detected. This can be caused by type errors, division by zero, out-of-bounds array access, or assert violations. This is acceptable in most situations. Note however that `assert()` should only be used to check invariants. Use `require()` for regular input checking. 
--------------------
In file: /tmp/44.sol:302

transactionIdsTemp[i]

--------------------
```
As you can see, some of the alarms show the function itself(not a line).
The line number it reports even indicates the start line of the function.

## Expected behavior
At least, the report points out the line number which can trigger the bug, not a whole function.

## Environment
- Mythril version: For Mythril, I used the docker version.
- OS: Ubuntu 16.04.5 LTS

**Comment by norhh** [comment]
This seems to be caused by return values which are arrays which internally does some additions,(probably raising a false positive,not yet verified).

We can just assume that the entire contract is shown when the issue is raised in the return statement or the parameters, As thats how it looks.

Also any mapping on the function return statement seems to show the entire function mapping,That should be fixed.

**Comment by audxo14** [comment]
Thanks for the reply.
I enjoy using this tool, and i hope it gets better and better

**Comment by norhh** [comment]
Hey @audxo14 , most of the problems in here are fixed in the latest version so I am closing the issue,  unfortunately the constructor mentioned in the issue seems to revert, as thats a different issue it is being addressed in here #920 .
